cmd/goal/messages.go

```diff
@@ -100,6 +100,9 @@ const (
 	errorMarshalingState           = "failed to encode state: %s"
 	errorApprovProgArgsRequired    = "Exactly one of --approval-prog or --approval-prog-raw is required"
 	errorClearProgArgsRequired     = "Exactly one of --clear-prog or --clear-prog-raw is required"
+	errorMissingBoxName            = "Box --name is required"
+	errorInvalidBoxName            = "Failed to parse box name %s. It must have the same form as app-arg."
+	errorBoxNameMismatch           = "Inputted box name %s does not match box name %s received from algod"
```

Comment:
This sounds like an error that only occurs in case we have bug. Looking a bit at our goal error messaging, I haven't found any analogues to this. I support the idea of _not_ sweeping this bug under the rug. I.e., just as is happening here, we admit that we made a mistake. I also don't think we should panic.

In indexer, we have ["shouldn't happen"](https://github.com/algorand/indexer/blob/a9412456d1e4bcaa7d667e038992c6874aa92141/api/error_messages.go#L31) type errors for such situations. I think such messaging would be useful, here, but don't have a strong opinion.

Reply:
I like this idea. I'll update the error message to be consistent with the ones in indexer.